File: dong99u/leetcode/2598.java
package leetcode.leetcode2597;

public class Solution {

	public static int findSmallestInteger(int[] nums, int value) {
        int[] count = new int[value];
        for (int x : nums) {
            int r = ((x % value) + value) % value;
            count[r]++;
        }

        int mex = 0;
        while (count[mex % value] > 0) {
            count[mex % value]--;
            mex++;
        }
        return mex;
    }
}

File: jiseok/LeetCode/2598.py
from collections import Counter

class Solution:
    def findSmallestInteger(self, nums: List[int], value: int) -> int:
        # 우선 0 이상의 정수로 압축
        for i in range(len(nums)):
            nums[i] %= value
        # Counter 사용하여 각 정수의 개수 세기
        nums_counter = Counter(nums)
        # 나올 수 있는 최대 MEX는 len(nums)
        for val in range(len(nums) + 1):
            # value로 나눈 나머지의 개수가 0개면 그 때의 val이 최대 MEX
            # 개수가 남아 있으면 줄여주고 val 증가
            if nums_counter[val % value]:
                nums_counter[val % value] -= 1
            else:
                return val